### PR TITLE
Make children optional in Link

### PR DESCRIPTION
**Pull request by bryancunningham-okta**
[OKTA-694627](https://oktainc.atlassian.net/browse/OKTA-694627)

## Summary
- Allow children to be optional in Link compoennt

## Testing & Screenshots

- [ ] I have confirmed this change with my designer and the Odyssey Design Team.

### CODE REVIEW

File: packages/odyssey-react-mui/src/Link.tsx [hunk 1]
```diff
@@ -27,7 +34,7 @@ export type LinkProps = {
   /**
    * The content within the Link
    */
-  children: React.ReactNode;
+  children?: ReactNode;
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/Link.tsx:37]: When passing `children`, do we have it setup to require an `ariaLabel`?

**Review by KevinGhadyani-Okta — COMMENTED** [review]
The changes are fine. I agree with what you did. I just wanna make sure we have our types correct for how we intend people to label the link when adding `children`.

File: packages/odyssey-react-mui/src/Link.tsx (hunk 1 above)
Comment (bryancunningham-okta) [reply packages/odyssey-react-mui/src/Link.tsx:37]: We do not. `children` can be anything here, or nothing now 😆 . `aria-label` is not usually required

**Comment by bryancunningham-okta** [comment]
> The changes are fine. I agree with what you did. I just wanna make sure we have our types correct for how we intend people to label the link when adding `children`.

In most cases, the label will be the text content of the children.

**Review by chrispulsinelli-okta — APPROVED** [review]

File: packages/odyssey-react-mui/src/Link.tsx (hunk 1 above)
Comment (KevinGhadyani-Okta) [reply packages/odyssey-react-mui/src/Link.tsx:37]: `aria-label` should be required if we have `children` passed. Now that it's properly optional, that's where we need to add a restriction that says if `children` is required, then `ariaLabel` is required. If `children` is optional, then we require `linkText`.

Comment (bryancunningham-okta) [reply packages/odyssey-react-mui/src/Link.tsx:37]: That doesn't really solve the problem we were looking to solve with this change. When interpolating strings of text using i18n `<Trans>` component, the text for the link is not available. It gets passed in as children by `<Trans>`. This causes TS errors with our current implementation. 
```js
<Trans
  i18nKey="your.key.name.here"
  components={[
  <Link
    key={0}
    href="https://whatever.com"
    target="_blank"
    rel="noopener noreferrer"
   />,
  ]}
/>
```

**Comment by bryancunningham-okta** [comment]
Closing this as we do not want to make children optional here

**Pull request closed without merging**